sdks/aperture-js/sdk/flow.ts

```diff
@@ -86,41 +96,90 @@
    * Sets the status of the flow.
    * @param status The status to set.
    */
-  SetStatus(status: FlowStatus) {
+  setStatus(status: FlowStatus) {
     this.status = status;
   }
 
   /**
-   * Sets the cached value for the flow.
-   * @param value The value to set.
-   * @param ttl The time-to-live for the cached value.
-   * @returns A promise that resolves to the response of the cache upsert operation.
+   * Sets the result cache entry for the flow.
+   * @param cacheEntry The cache entry to set.
+   * @returns A promise that resolves to the response of the key upsert operation.
    */
-  async SetCachedValue(value: Buffer, ttl: Duration) {
-    if (!this.cacheKey) {
-      return Promise.reject(new Error("No cache key"));
-    }
+  async setResultCache(cacheEntry: CacheEntry) {
+    return new Promise<KeyUpsertResponse>((resolve) => {
+      if (!this.resultCacheKey) {
+        const resp = new _KeyUpsertResponse(new Error("No cache key"));
+        resolve(resp);
+        return;
+      }
+      let cacheUpsertRequest: CacheUpsertRequest = {
+        controlPoint: this.controlPoint,
+        resultCacheEntry: {
+          key: this.resultCacheKey,
+          value: cacheEntry.value,
+          ttl: cacheEntry.ttl,
+        },
+      };
+      this.fcsClient.CacheUpsert(
+        cacheUpsertRequest,
+        this.grpcCallOptions,
+        (err, res) => {
+          if (err) {
+            const resp = new _KeyUpsertResponse(err);
+            resolve(resp);
+            return;
+          }
+          if (!res) {
+            const resp = new _KeyUpsertResponse(
+              new Error("No cache upsert response"),
+            );
+            resolve(resp);
+            return;
+          }
+          const resp = new _KeyUpsertResponse(
+            convertCacheError(res.resultCacheResponse?.error),
+          );
+          resolve(resp);
+        },
+      );
+    });
+  }
 
-    const key = this.cacheKey;
-    return new Promise<SetCachedValueResponse | undefined>((resolve) => {
-      const cacheUpsertRequest: CacheUpsertRequest = {
+  /**
+   * Sets a state cache entry for the flow.
+   * @param key The key of the cache entry to set.
+   * @param cacheEntry The cache entry to set.
+   * @returns A promise that resolves to the response of the key upsert operation.
+   */
+  async setStateCache(key: string, cacheEntry: CacheEntry) {
+    return new Promise<KeyUpsertResponse>((resolve) => {
+      let cacheUpsertRequest: CacheUpsertRequest = {
         controlPoint: this.controlPoint,
-        key: key,
-        value: value,
-        ttl: ttl,
+        stateCacheEntries: {
+          key: {
+            value: cacheEntry.value,
+            ttl: cacheEntry.ttl,
+          },
```

Comment:
The key in state cache entries is optional since they are keyed by it in the stateCacheEntries map that contains them in the CacheUpsertRequest protobuf structure